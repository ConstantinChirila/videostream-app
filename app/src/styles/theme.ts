export const theme = {
  colors: {
    primary: "#FF286A",
    secondary: "#5000B4",
    dark: "#172029",
    light: "#F5F6F8",
  },
} as any;
