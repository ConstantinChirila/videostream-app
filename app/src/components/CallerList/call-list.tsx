import React, { ReactElement } from "react";
import { StyledContainer, StyledButton } from "./call-list.styles";

type TProps = {
  users: {};
  yourID: string;
  callPeer: (id: string) => void;
};

export function CallerList({ users, yourID, callPeer }: TProps): ReactElement {
  return (
    <StyledContainer>
      <h3>Active Users</h3>
      {Object.keys(users).map((key) => {
        if (key === yourID) {
          return null;
        }

        return (
          <StyledButton key={key} onClick={() => callPeer(key)}>
            <CallIcon /> {key}
          </StyledButton>
        );
      })}
    </StyledContainer>
  );
}

function CallIcon() {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      strokeWidth="2"
      stroke="currentColor"
      fill="none"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M0 0h24v24H0z" stroke="none" />
      <path d="M5 4h4l2 5-2.5 1.5a11 11 0 005 5L15 13l5 2v4a2 2 0 01-2 2A16 16 0 013 6a2 2 0 012-2M15 7a2 2 0 012 2M15 3a6 6 0 016 6" />
    </svg>
  );
}
