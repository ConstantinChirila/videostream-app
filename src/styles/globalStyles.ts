import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;1,500&display=swap');

  html {
    font-size: 10px;
  }
  
  body {
    font-family: 'Roboto', sans-serif;
    color: ${theme.colors.dark};
    background-color: ${theme.colors.light};
    font-size: 1.6rem
  }
`;
