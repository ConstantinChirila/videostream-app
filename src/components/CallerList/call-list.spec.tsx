import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import { CallerList } from "./call-list";

const mockUsers = { test: "test" };
const mockID = "testID";
const mockCallPeer = jest.fn();

describe("CallAlert", () => {
  it("should render the component", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <CallerList users={mockUsers} yourID={mockID} callPeer={mockCallPeer} />
      </ThemeProvider>
    );
    const user = screen.getByText(mockUsers.test);
    expect(user).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not show any users", () => {
    render(
      <ThemeProvider theme={theme}>
        <CallerList users={{}} yourID={mockID} callPeer={mockCallPeer} />
      </ThemeProvider>
    );
    const user = screen.getByText("No active users.");
    expect(user).toBeInTheDocument();
  });
  it("should call the user on click", () => {
    const mockFn = jest.fn();
    render(
      <ThemeProvider theme={theme}>
        <CallerList users={mockUsers} yourID={mockID} callPeer={mockFn} />
      </ThemeProvider>
    );
    const regex = new RegExp(mockUsers.test, "i");
    fireEvent.click(screen.getByText(regex));
    expect(mockFn).toHaveBeenCalledWith(mockUsers.test);
  });
});
