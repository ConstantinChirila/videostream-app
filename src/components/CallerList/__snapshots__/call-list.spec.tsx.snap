// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallAlert should render the component 1`] = `
<div
  class="sc-bdfBwQ ccowex"
>
  <h3>
    Active Users
  </h3>
  <button
    class="sc-gsTCUz doTqLt"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        stroke="none"
      />
      <path
        d="M5 4h4l2 5-2.5 1.5a11 11 0 005 5L15 13l5 2v4a2 2 0 01-2 2A16 16 0 013 6a2 2 0 012-2M15 7a2 2 0 012 2M15 3a6 6 0 016 6"
      />
    </svg>
     
    test
  </button>
</div>
`;
